onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc

```diff
@@ -866,6 +865,8 @@ Status WebGpuExecutionProvider::OnRunEnd(bool /* sync_stream */, const onnxrunti
     return context_.PopErrorScope();
   }
 
+  context_.OnRunEnd();
+
```

Comment:
This need to be put before line 864, otherwise it will not be executed if validation level is equal to or greater than basic.

The code is a little bit error prone. suggest to modify to:
```
  if (context_.ValidationMode() >= ValidationMode::Basic) {
    return context_.PopErrorScope();
  } else {
    return Status::OK();
  }
```

The modified code is much less likely to make same mistake in future.